btree/btree.c

```diff
@@ -11863,6 +11869,142 @@ static void ut_btree_truncate(void)
+	void                       *v_ptr           = &value;
+	int                         rc;
+	struct m0_buf               buf;
+	uint32_t                    rnode_sz        = 1024;
+	struct m0_fid               fid             = M0_FID_TINIT('b', 0, 1);
+	uint32_t                    rnode_sz_shift;
+	struct m0_btree_rec         rec             = {
+			    .r_key.k_data = M0_BUFVEC_INIT_BUF(&k_ptr, &ksize),
+			    .r_val        = M0_BUFVEC_INIT_BUF(&v_ptr, &vsize),
+			};
+	struct cb_data              put_data;
+	/**
+	 * In this UT, we are testing the functionality of LRU list purge and
+	 * be-allocator with chunk align parameter.
+	 *
+	 * 1. Allocate and fill up the btree with multiple nodes.
+	 * 2. Verify the size increase in memory by adding breakpoint.
+	 * 3. Use the m0_btree_lrulist_purge() to reduce the size by freeing up
+	 *    the unused nodes present in LRU list.
+	 * 4. Verify the reduction in size.
+	 */
+	M0_ENTRY();
+
+	btree_ut_init();
+	mem_init = sysconf(_SC_AVPHYS_PAGES) * sysconf(_SC_PAGESIZE);
+	printf("Mem Init (%"PRId64").\n",mem_init);
+	sleep(15);
+
+	M0_ASSERT(rnode_sz != 0 && m0_is_po2(rnode_sz));
+	rnode_sz_shift = __builtin_ffsl(rnode_sz) - 1;
+	cred = M0_BE_TX_CB_CREDIT(0, 0, 0);
+	m0_be_allocator_credit(NULL, M0_BAO_ALLOC_ALIGNED, rnode_sz,
+			       rnode_sz_shift, &cred);
+	m0_btree_create_credit(&bt, &cred, 1);
+
+	/** Prepare transaction to capture tree operations. */
+	m0_be_ut_tx_init(tx, ut_be);
+	m0_be_tx_prep(tx, &cred);
+	rc = m0_be_tx_open_sync(tx);
+	M0_ASSERT(rc == 0);
+
+	/** Create temp node space and use it as root node for btree */
+	buf = M0_BUF_INIT(rnode_sz, NULL);
+	M0_BE_ALLOC_CHUNK_ALIGN_BUF_SYNC(&buf, rnode_sz_shift, seg, tx);
+	rnode = buf.b_addr;
+
+	rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op, m0_btree_create(rnode, rnode_sz,
+							     &bt,
+							     &b_op, &btree, seg,
+							     &fid, tx, NULL));
+	M0_ASSERT(rc == M0_BSC_SUCCESS);
+	m0_be_tx_close_sync(tx);
+	m0_be_tx_fini(tx);
+
+	tree = b_op.bo_arbor;
+
+	cred = M0_BE_TX_CB_CREDIT(0, 0, 0);
+	m0_btree_put_credit(tree, 1, ksize, vsize, &cred);
+
+	put_data.key       = &rec.r_key;
+	put_data.value     = &rec.r_val;
+
+	ut_cb.c_act        = btree_kv_put_cb;
+	ut_cb.c_datum      = &put_data;
+
+	for (i = 1; i <= rec_count; i++) {
+		int      k;
+
+		key = m0_byteorder_cpu_to_be64(i);
+		for (k = 0; k < ARRAY_SIZE(value); k++)
+			value[k] = key;
+
+		m0_be_ut_tx_init(tx, ut_be);
+		m0_be_tx_prep(tx, &cred);
+		rc = m0_be_tx_open_sync(tx);
+		M0_ASSERT(rc == 0);
+
+		rc = M0_BTREE_OP_SYNC_WITH_RC(&kv_op,
+					      m0_btree_put(tree, &rec,
+							   &ut_cb,
+							   &kv_op, tx));
+		M0_ASSERT(rc == 0 && put_data.flags == M0_BSC_SUCCESS);
+		m0_be_tx_close_sync(tx);
+		m0_be_tx_fini(tx);
+	}
+
+	mem_after_alloc = sysconf(_SC_AVPHYS_PAGES) * sysconf(_SC_PAGESIZE);
+	mem_increased = mem_init - mem_after_alloc;
+	printf("Mem After Alloc (%"PRId64") || Mem Increase (%"PRId64").\n",
+	       mem_after_alloc, mem_increased);
+	sleep(15);
```

Comment:
Remove all these sleep in the code as you are not using this pause to verify the memory status from outside the ut.

Reply:
Done.